### PR TITLE
Document project layout

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: doc/contributor/project-layout.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Project Layout
+
+Each folder in the TruffleRuby project is described below:
+
+## `/bench/`
```
Comment (eregon) [inline doc/contributor/project-layout.md]: `##` renders huge on GitHub.
Maybe a lower-level title or just a list would be more readable?

File: doc/contributor/project-layout.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Files for running benchmarks. See the `jt benchmark` command.
+
+## `/bin/`
```
Comment (eregon) [inline doc/contributor/project-layout.md]: The leading slash looks strange to me, I would prefer `bin/` as it's not a path from the system root directory.

File: doc/contributor/project-layout.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Scripts used to for common Ruby commands. E.g. `ruby`, `gem`.
+
+## `/doc/`
+Project documentation for users and contributors.
+
+## `/lib/`
+Contains the stdlib, core, and TruffleRuby specific Ruby and C libraries.
```
Comment (eregon) [inline doc/contributor/project-layout.md]: TruffleRuby-specific

Comment (eregon) [reply doc/contributor/project-layout.md]: It doesn't contain `core`.

File: doc/contributor/project-layout.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+TruffleRuby logo assets.
+
+## `/mx.truffleruby/`
+Files used by the `mx` build system.
+
+## `/spec/`
+Ruby Spec suite, tags, runner and additional TruffleRuby specific specs.
+
+## `/src/`
+Sources and tests implementing the Ruby lanaguage, core, C extension API, and stdlib.
```
Comment (eregon) [inline doc/contributor/project-layout.md]: `language`.
No stdlib in `src/`.

Comment (eregon) [reply doc/contributor/project-layout.md]: There is only a couple Java tests under `src/`, I would skip in this description or precise it's Java tests.

File: doc/contributor/project-layout.md (hunk 1 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed to `###`.

File: doc/contributor/project-layout.md (hunk 2 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed, removed leading slashes.

File: doc/contributor/project-layout.md (hunk 3 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed and fixed.

File: doc/contributor/project-layout.md (hunk 4 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed these, thanks.

File: doc/contributor/project-layout.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Project Layout
+
+Each folder in the TruffleRuby project is described below:
+
+### `bench/`
+Files for running benchmarks. See the `jt benchmark` command.
+
+### `bin/`
+Scripts used to for common Ruby commands. E.g. `ruby`, `gem`.
```
Comment (eregon) [inline doc/contributor/project-layout.md]: `to for` => `for`

File: doc/contributor/project-layout.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+Contains the stdlib, and TruffleRuby-specific Ruby and C libraries.
+
+### `logo/`
+TruffleRuby logo assets.
+
+### `mx.truffleruby/`
+Files used by the `mx` build system.
+
+### `spec/`
+Ruby Spec suite, tags, runner and additional TruffleRuby specific specs.
```
Comment (eregon) [inline doc/contributor/project-layout.md]: `TruffleRuby-specific`

File: doc/contributor/project-layout.md (hunk 5 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed.

File: doc/contributor/project-layout.md (hunk 6 above)
Comment (bjfish) [reply doc/contributor/project-layout.md]: Fixed.

**Pull request merged**